### PR TITLE
Fix AuthLogic shutdown race

### PR DESCRIPTION
**Pull request by sergey-chernikov**

### CODE REVIEW

File: BlocksettleNetworkingLib/AuthAddressLogic.cpp [hunk 1]
```diff
@@ -1135,6 +1213,73 @@ bool AuthAddressLogic::isValid(
...
+            return;
+         }
+
+         if (!pathState.revokedPaths_.empty()) {
+            //validation output was spent by user
+            cb(addr, AddressVerificationState::Revoked);
+            return;
+         }
+         //address has history and no validation outputs
+         cb(addr, AddressVerificationState::Tainted);
```
Comment (goatpig) [inline BlocksettleNetworkingLib/AuthAddressLogic.cpp:1274]: this should return after firing the callback

File: BlocksettleNetworkingLib/AuthAddressLogic.cpp [hunk 2]
```diff
@@ -450,26 +448,53 @@ bool AuthAddressValidator::goOnline(const ResultCb &cb)
...
-   }
+   //grab all txouts
+   lambdas_->getOutpointsForAddresses(addrVec, cb, topBlock_, zcIndex_);
+}
 
+unsigned AuthAddressValidator::update(const OutpointBatch &batch)
+{
+   std::unique_lock<std::mutex> lock(updateMutex_);
+   if (!lambdas_ || stopped_) {
+      std::cout << "stopped 1\n";
```
Comment (goatpig) [inline BlocksettleNetworkingLib/AuthAddressLogic.cpp:488]: left over print statement

File: BlocksettleNetworkingLib/AuthAddressLogic.cpp [hunk 3]
```diff
@@ -450,26 +448,53 @@ bool AuthAddressValidator::goOnline(const ResultCb &cb)
...
+      return UINT32_MAX;
+   }
    unsigned opCount = 0;
-   for (auto& outpointPair : batch.outpoints_) {
+   for (const auto &outpointPair : batch.outpoints_) {
       if (stopped_) {
-         return 0;
+         std::cout << "stopped 2\n";
```
Comment (goatpig) [inline BlocksettleNetworkingLib/AuthAddressLogic.cpp:494]: same as above

File: BlocksettleNetworkingLib/AuthAddressLogic.cpp (hunk 1 above)
Comment (sergey-chernikov) [reply BlocksettleNetworkingLib/AuthAddressLogic.cpp:1272]: Thank you, fixed

**Pull request merged**